### PR TITLE
Theme Toggler

### PR DESCRIPTION
**Pull request by rileyhawk1417**
## Theme Toggle Button for AppFlowy

- Theme toggle allows the user to switch between light and dark theme with ease.

### CODE REVIEW

File: frontend/app_flowy/lib/startup/tasks/application_widget.dart [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import 'package:app_flowy/startup/startup.dart';
+import 'package:app_flowy/workspace/presentation/theme/themeProvider.dart';
```
Comment (appflowy) [inline frontend/app_flowy/lib/startup/tasks/application_widget.dart:2]: theme_provider.dart would be better?

File: frontend/app_flowy/lib/startup/tasks/application_widget.dart [hunk 2]
```diff
@@ -39,31 +39,48 @@ class ApplicationWidget extends StatelessWidget {
...
+        );
+      });
+}
+
+class App_Theme extends ChangeNotifier {
+  ThemeType get theme => _theme;
+  ThemeType _theme = ThemeType.light;
+
+  set theme(ThemeType value) {
+    _theme = value;
```
Comment (appflowy) [inline frontend/app_flowy/lib/startup/tasks/application_widget.dart:79]: Compare the value is the same as _theme would be better

File: frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart [hunk 3]
```diff
@@ -54,6 +60,49 @@ class MenuUser extends StatelessWidget {
     );
   }
 
+  Widget _renderThemeToggle(BuildContext context) {
+    final themeProvider = Provider.of<ThemeProvider>(context);
+    final provider = Provider.of<ThemeProvider>(context, listen: false);
```
Comment (appflowy) [inline frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart:65]: Unuse?

File: frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart [hunk 4]
```diff
@@ -54,6 +60,49 @@ class MenuUser extends StatelessWidget {
...
+    final theme = context.watch<AppTheme>();
+    return Switch.adaptive(
+      value: themeProvider.isDarkMode,
+      onChanged: (value) {
+        final provider = Provider.of<ThemeProvider>(context, listen: false);
+        provider.toggleTheme(value);
+        print(themeProvider.isDarkMode);
+      },
+    );
+    // return Material(
```
Comment (appflowy) [inline frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart:75]: Comment out code should not be submitted

File: frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart [hunk 5]
```diff
@@ -54,6 +60,49 @@ class MenuUser extends StatelessWidget {
...
+    //         print(themeProvider.isDarkMode);
+    //         print(themeProvider.themeMode.name);
+    //         print(theme.textColor);
+    //       }),
+    // ));
+  }
+
+  Widget _renderDarkMode(BuildContext context) {
+    return Tooltip(
+      message: "Set to Dark Mode",
```
Comment (appflowy) [inline frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart:94]: Language support. 

e.g. LocaleKeys.subscribeNewsletterText.tr(),

File: frontend/app_flowy/lib/startup/tasks/application_widget.dart (hunk 2 above)
Comment (rileyhawk1417) [reply frontend/app_flowy/lib/startup/tasks/application_widget.dart:79]: Forgot to delete that, will review it maybe could use it somewhere.

File: frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart (hunk 5 above)
Comment (rileyhawk1417) [reply frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart:94]: Noted will add that in.

File: frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart (hunk 4 above)
Comment (rileyhawk1417) [reply frontend/app_flowy/lib/workspace/presentation/widgets/menu/widget/menu_user.dart:75]: Will submit the final edits after seeing which code works better.

**Comment by rileyhawk1417** [comment]
I finally managed to get the Toggle button working after some tweaks. Please let me know if there are any changes to be done.

**Comment by appflowy** [comment]
Issue fixed. Will be submitted later.

The reason for the data corruption is the document data was written even if the data already exists in the local disk. Which causes duplicate data issues.

**Comment by MikeWallaceDev** [comment]
I had to rebase this PR and submitted a new one. Please see PR #206

**Pull request closed without merging**